### PR TITLE
add label_names to make_csv_dataset_v2

### PR DESCRIPTION
**Pull request by DachuanZhao**
[https://github.com/tensorflow/tensorflow/issues/51571](https://github.com/tensorflow/tensorflow/issues/51571)

### CODE REVIEW

**Comment by gbaned** [comment]
@DachuanZhao  Can you please address PyLint errors? Thanks!

**Comment by DachuanZhao** [comment]
> address PyLint errors

How could I address PyLint errors in vscode ? Is there a guide or a document ?

File: tensorflow/python/data/experimental/ops/readers.py [hunk 1]
```diff
@@ -463,9 +468,11 @@ def make_csv_dataset_v2(
   Returns:
     A dataset, where each element is a (features, labels) tuple that corresponds
     to a batch of `batch_size` CSV rows. The features dictionary maps feature
-    column names to `Tensor`s containing the corresponding column data, and
-    labels is a `Tensor` containing the column data for the label column
-    specified by `label_name`.
+    column names to `Tensor`s containing the corresponding column data. If
```
Comment (aaudiber) [inline tensorflow/python/data/experimental/ops/readers.py:471]: Is the original doc accurate? IIUC there will only be a `labels` part of the tuple if `label_name` or `label_names` is set.

Perhaps we could say something like:

"A dataset, where each element is either a dictionary of features or, if `label_name` or `label_names` are specified, a (features, labels) tuple."

**Review by aaudiber — COMMENTED** [review]
Thanks! Can you please also add unit tests in https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/data/experimental/kernel_tests/make_csv_dataset_test.py?

**Comment by gbaned** [comment]
@DachuanZhao Can you please check @aaudiber's comments and keep us posted ? Thanks!

File: tensorflow/python/data/experimental/ops/readers.py (hunk 1 above)
Comment (DachuanZhao) [reply tensorflow/python/data/experimental/ops/readers.py:471]: There is a problem with the return format design. I think the general design is returning a ```(collections.OrderedDict() , collections.OrderedDict())``` tuple which is compatible with multiple label names. But it return a ```(collections.OrderedDict() , Tensor )``` now ...

**Comment by DachuanZhao** [comment]
> Thanks! Can you please also add unit tests in https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/data/experimental/kernel_tests/make_csv_dataset_test.py?

Get it. I will add the unit test.

**Comment by gbaned** [comment]
@DachuanZhao  Any update on this PR? Please. Thanks!

**Comment by gbaned** [comment]
@DachuanZhao Any update on this PR? Please. Thanks!

**Pull request closed without merging**